src/test/java/com/kodilla/ecommercee/repositoryDao/CartRepositoryTestSuit.java

```diff
@@ -0,0 +1,200 @@
+        product1 = new Product("Glosniki","czarne",23);
+        product2 = new Product("Telefon","bezprzewodowy",40);
+        product3 = new Product("Pokrowiec","przezroczysty",15);
+
+        listOfProducts1 = new ArrayList<>();
+        listOfProducts2 = new ArrayList<>();
+        listOfProducts3 = new ArrayList<>();
+
+        cart1 = new Cart();
+        cart1.setProductsList(listOfProducts1);
+        cart2 = new Cart();
+        cart2.setProductsList(listOfProducts2);
+        cart3 = new Cart();
+        cart3.setProductsList(listOfProducts3);
+    }
+    @Transactional
+    @Test
+    public void testCartDaoSaveAndFindAll(){
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        //When
+        List<Cart> listOfCarts = cartRepository.findAll();
+        //Then
+        Assert.assertEquals(3,listOfCarts.size());
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart2.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartDaoSaveAndFindById(){
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        Long IdCart1 = cart1.getId();
+        //When
+        Optional<Cart> cartFound = cartRepository.findById(IdCart1);
+        //Then
+        Assert.assertEquals(IdCart1,cartFound.get().getId());
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart2.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+    @Transactional
+    @Test
+    public void testCartDaoDeleteById(){
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        Long cart2Id = cart2.getId();
+        //When
+        cartRepository.deleteById(cart2Id);
+        List<Cart> cartList = cartRepository.findAll();
+        //Then
+        Assert.assertFalse(cartList.contains(cart2));
+        //Clean Up
+        try{
+            cartRepository.deleteById(cart1.getId());
+            cartRepository.deleteById(cart3.getId());
+        }catch (Exception e){
+            LOGGER.error("Clean-up process failed.", e.getMessage(), e);
+        }
+    }
+
+    @Transactional
+    @Test
+    public void testCartOneOnOneWithOrder(){
+        //Given
+        Order order = new Order();
+        cart1.setOrder(order);
+        //When
+        cartRepository.save(cart1);
+        Optional<Cart> orderId = cartRepository.findById(cart1.getOrder().getOrderId());
```

Comment:
Co tu chcesz sprawdzic? Chcesz  pobrac Cart czy Order? Jezeli Cart to tak:     
Optional<Cart> orderId = cartRepository.findById(cart1.getId());
        //Then
        Assert.assertNotEquals(zeroLong,orderId.get().getId());

Reply:
@erwinMuz chialem tutaj wyciagnac orderId z repo i sprawdzic czy zapisalo jakas wartosc

Reply:
Tutaj nadal jest blad. Wyciagasz Cart po id z Order. Nastepnie porownujesz obiekt Optional z zeroLong.
Jezeli chcesz wyciagnac Order uzyj orderRepository albo wytarczy tak:
Assert.assertNotEquals(zeroLong, cart1.getOrder().getOrderId());

Reply:
ok @erwinMuz Dzieki w koncu zrobilem mam nadzieje ze teraz bedzie zielone swiatlo dzieki